lib/cartopy/mpl/geoaxes.py

```diff
@@ -1281,7 +1281,8 @@ def imshow(self, img, *args, **kwargs):
                           y0 - eps <= extent[3] <= y1 + eps))
 
         if (transform is None or transform == self.transData or
-                same_projection and inside_bounds):
+                same_projection and inside_bounds and
+                kwargs.get("regrid_shape", None) is None):
```

Comment:
This would forward on a manually-specified `regrid_shape=None` onto Matplotlib, no?